Undefined reference to aeron::ExclusivePublication::channelStatus()

In `ExclusivePublication.h` `channelStatus()` is defined as both const and non const method:

- https://github.com/real-logic/aeron/blob/master/aeron-client/src/main/cpp/ExclusivePublication.h#L305
- https://github.com/real-logic/aeron/blob/master/aeron-client/src/main/cpp/ExclusivePublication.h#L600

But it is implemented only as const method in `ExclusivePublication.cpp`: https://github.com/real-logic/aeron/blob/master/aeron-client/src/main/cpp/ExclusivePublication.cpp#L96

This can lead to linker errors like this one:

```
undefined reference to 'aeron::ExclusivePublication::channelStatus()'
```

This was introduced in: https://github.com/real-logic/aeron/commit/d90f3c00adafc653191f3c054ee74d59c58058b3#diff-42799ad76422e95b191febdf62b38699

--- a/aeron-client/src/main/cpp/ExclusivePublication.h
+++ b/aeron-client/src/main/cpp/ExclusivePublication.h
@@ -592,13 +592,6 @@ public:
      */
     void removeDestination(const std::string &endpointChannel);
 
-    /**
-     * Get the status for the channel of this {@link ExclusivePublication}
-     *
-     * @return status code for this channel
-     */
-    std::int64_t channelStatus();
-
     /// @cond HIDDEN_SYMBOLS
     inline void close()
     {

